fractal.js

```diff
@@ -0,0 +1,51 @@
+(function () {
+  // Create Canvas
+  var myCanvas = document.createElement("canvas");
+  myCanvas.width = document.documentElement.clientWidth;
+  myCanvas.height = document.documentElement.clientHeight;
+  // myCanvas.width = 1200;
+  // myCanvas.height = 300;
+
+  document.body.appendChild(myCanvas);
+  var ctx = myCanvas.getContext("2d");
+
+  // Start drawing
+  function checkIfBelongsToMandelbrotSet(x, y) {
+    var realComponentOfResult = x;
+    var imaginaryComponentOfResult = y;
+    var maxIterations = 100;
+    for (var i = 0; i < maxIterations; i++) {
+      var tempRealComponent = realComponentOfResult * realComponentOfResult
+        - imaginaryComponentOfResult * imaginaryComponentOfResult
+        + x;
+      var tempImaginaryComponent = 2 * realComponentOfResult * imaginaryComponentOfResult
+        + y;
+      realComponentOfResult = tempRealComponent;
+      imaginaryComponentOfResult = tempImaginaryComponent;
+
+      // Return a number as a percentage
+      if (realComponentOfResult * imaginaryComponentOfResult > 5)
+        return (i / maxIterations * 100);
+    }
+    return 0;   // Return zero if in set        
+  }
+
+  var magnificationFactor = 500;
+  var panX = Math.random() * 3;
+  var panY = Math.random() * 3;
+  console.log(panX, panY)
+  for (var x = 0; x < myCanvas.width; x++) {
+    for (var y = 0; y < myCanvas.height; y++) {
+      var belongsToSet =
```

Comment:
here youre nesting for loops which has O(n2) runtime that you should try to avoid if possible